packages/webview_flutter/webview_flutter_wkwebview/lib/src/ui_kit/ui_kit_api_impls.dart

```diff
@@ -114,3 +125,53 @@ class UIViewHostApiImpl extends UIViewHostApi {
     return setOpaque(instanceManager.getIdentifier(instance)!, opaque);
   }
 }
+
+/// Flutter api implementation for [UIScrollViewDelegate].
+class UIScrollViewDelegateFlutterApiImpl
+    extends UIScrollViewDelegateFlutterApi {
+  /// Constructs a [UIScrollViewDelegateFlutterApiImpl].
+  UIScrollViewDelegateFlutterApiImpl({InstanceManager? instanceManager})
+      : instanceManager = instanceManager ?? NSObject.globalInstanceManager;
+
+  /// Maintains instances stored to communicate with native language objects.
+  final InstanceManager instanceManager;
+
+  UIScrollViewDelegate _getDelegate(int identifier) {
+    return instanceManager.getInstanceWithWeakReference(identifier)!;
+  }
+
+  @override
+  void scrollViewDidScroll(int identifier, int uiScrollViewIdentifier) {
+    final void Function(UIScrollView)? callback =
+        _getDelegate(identifier).scrollViewDidScroll;
+    final UIScrollView? uiScrollView = instanceManager
+        .getInstanceWithWeakReference(uiScrollViewIdentifier) as UIScrollView?;
+    if (uiScrollView != null && callback != null) {
```

Comment:
Instead of silently failing when `uiScrollView` is null, this should throws an assertion error with 

```dart
assert(uiScrollView != null);
```